enumerable.rb

```diff
@@ -0,0 +1,115 @@
+module Enumerable
+  def my_each
+    to_a.length.times { |i| yield self[i] }
+    self
+  end
+
+  def my_each_with_index
+    to_a.my_each { |each| yield each, index(each) }
+    self
+  end
```

Comment:
## my_each_with_index

 - [ ] returns an enumerator if no block is given
```ruby
array.my_each_with_index
```
Expected: Enumerator
got:LocalJumpError:
       no block given (yield)

- [ ] when a block is given when self is an array behaves like #each_with_index calls the given block once for each element in self
```ruby
my_each_with_index_output
```
Expected: "Num: 3, idx: 0\nNum: 5, idx: 1\nNum: 5, idx: 2\nNum: 7, idx: 3\nNum: 1, idx: 4\nNum: 1, idx: 5\nNum:...: 8, idx: 94\nNum: 8, idx: 95\nNum: 4, idx: 96\nNum: 4, idx: 97\nNum: 2, idx: 98\nNum: 5, idx: 99\n"
got: "Num: 3, idx: 0\nNum: 5, idx: 1\nNum: 5, idx: 1\nNum: 7, idx: 3\nNum: 1, idx: 4\nNum: 1, idx: 4\nNum:...Num: 8, idx: 7\nNum: 8, idx: 7\nNum: 4, idx: 23\nNum: 4, idx: 23\nNum: 2, idx: 12\nNum: 5, idx: 1\n"

- [ ] when a block is given when self is a range returns the range itself after calling the given block once for each element in self
```ruby
range.my_each_with_index(&block)
```
got: NoMethodError:
       undefined method `index' for 5..50:Range

- [ ] when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
```ruby
enum.my_each_with_index(&block)
```
got: NoMethodError:
       undefined method `index' for 5..50:Range

- [ ] when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self
```ruby
my_each_with_index_output
```
expected: "Num: [:a, 1], idx: 0\nNum: [:b, 2], idx: 1\nNum: [:c, 3], idx: 2\nNum: [:d, 4], idx: 3\nNum: [:e, 5], idx: 4\n"
            got: "Num: [:a, 1], idx: \nNum: [:b, 2], idx: \nNum: [:c, 3], idx: \nNum: [:d, 4], idx: \nNum: [:e, 5], idx: \n"